### ISSUE TITLE
Support for Apache Commons Compress Version of Pack200

### ISSUE THREAD

**Issue by martinhickson** [AdoptOpenJDK/IcedTea-Web#936]
Since the JDK's native pack200 implementation has been removed from newer JDKs, it would be great if this implementation could be supported.

**Comment by puck** [comment]
Hey, just coming to support this request as the service processors on older servers may use Pack200 for their JAR files and they won't work on newer JDKs. Here's a write up of someone's findings and that they've had to fall back to an ancient JDK version: https://forums.debian.net/viewtopic.php?t=158747

**Comment by karianna** [comment]
It does look like the original poster managed to get past this.  @hendrikebbers may be able to help in the future though.

**Comment by puck** [comment]
Yup, by updating a fork of the Pack200 code from JDK11 to compile with JDK17 and updating IcedTea to use it. This be great have back in the official release.

I started to add in support for the Pack200 class that is in Apache Commons Compress (which appears to have been Martin's initial approach as well) and it is a game of whack-a-mole to fix/add things which I haven't finished.

**Comment by martinhickson** [comment]
See also: https://github.com/martinhickson/pack200/wiki, which relates to a fork of the pack200 project, where changes to attribute handling in the ClassWriter class (see the commit history), achieved full pack200 binary compatibility with the JDK 11 packer.
